RecoLocalTracker/SiStripClusterizer/src/ThreeThresholdAlgorithm.cc

```diff
@@ -98,7 +101,7 @@ inline void ThreeThresholdAlgorithm::endCandidate(State& state, T& out) const {
 }
 
 inline bool ThreeThresholdAlgorithm::candidateAccepted(State const& state) const {
-  return (!state.candidateLacksSeed &&
+  return (!state.candidateLacksSeed && state.ADCs.size() <= MaxClusterSize &&
```

Comment:
For compatibility with the CPU implementation, which has a maximum cluster size so that the cluster arrays can be pre-allocated with a fixed size.  This change is a minor optimization which lets the legacy clusterizer terminate early if the max cluster size is exceeded, which can greatly reduce memory allocations for ridiculously large clusters.